### DISCUSSION TITLE
Can I get repository by repository name

### DISCUSSION THREAD

**Discussion by imcm7** [mikro-orm/mikro-orm#4758] [General]
I need here entity or entity name `em.getRepository('entity_name');` but I have only repository name `UserRepository` and UserRepository class`[class UserRepository extends SqlEntityRepository]`.

Can I get repository by repository name like`em.getRepositoryByName(UserRepository);` ?
Or `em.getRepository('entity name', 'entity class', 'repository name', 'repository class');`

Its strange to not get repository by it's name.

**Comment by imcm7** [comment]
Or I can add a static property to entityName repository class and and to get entity name from it?

**Reply by B4nan** [reply]
Yes, if it's a custom repository, you can add a static property to it.
